src/Lumi/Components2/Button.purs

```diff
@@ -0,0 +1,145 @@
+  lumiComponent "Button" defaults render
+  where
+  lumiButtonElement ::
+    forall attrs attrs_.
+    Union attrs attrs_ ( | R.Props_button ) =>
+    ReactComponent
+      { className :: String
+      , "aria-label" :: Nullable.Nullable String
+      | attrs
+      }
+  lumiButtonElement = R.unsafeCreateDOMComponent "button"
+
+  lumiButtonLinkElement ::
+    forall attrs attrs_.
+    Union attrs attrs_ ( | R.Props_a ) =>
+    ReactComponent
+      { className :: String
+      , "aria-label" :: Nullable.Nullable String
+      | attrs
+      }
+  lumiButtonLinkElement = R.unsafeCreateDOMComponent "a"
+
+  defaults :: Record ButtonProps
+  defaults =
+    { accessibilityLabel: mempty
+    , onPress: mempty
+    , size: Medium
+    , testId: mempty
+    , type: mempty
+    , kind: Primary
+    , state: Enabled
+    , color: Nothing
+    , content: mempty
+    }
+
+  render props = React.do
+    theme <- useTheme
+    pure
+      if props.type == "link" then
+        E.element lumiButtonLinkElement
+          { "aria-label": Nullable.toNullable props.accessibilityLabel
+          , children: props.content
+          , className: props.className
+          , css: toCSS theme props (Styles.Button.button props.color props.kind props.state props.size)
+          , onClick: capture_ props.onPress
+          , role: "button"
+          , _data:
+            fromHomogeneous
+              { testid: fold props.testId
+              }
+          }
+      else
+        E.element lumiButtonElement
+          { "aria-label": Nullable.toNullable props.accessibilityLabel
+          , children: props.content
+          , className: props.className
+          , css: toCSS theme props (Styles.Button.button props.color props.kind props.state props.size)
+          , onClick: capture_ props.onPress
+          , type: props.type
+          , _data:
+            fromHomogeneous
+              { testid: fold props.testId
+              , size: show props.size
+              , loading:
+                show case props.state of
+                  Enabled -> false
+                  Disabled -> false
+                  Loading -> true
+              -- , color: un ColorName props.color
+              }
+          , disabled:
+            case props.state of
+              Enabled -> false
+              Disabled -> true
+              Loading -> false
+          }
+    where
+    children =
+      if Array.length props.content == 0 then
+        [ R.text invisibleSpace ] -- preserves button size when content is empty
+      else
+        props.content
+
+primary :: PropsModifier ButtonProps
+primary = identity
+
+secondary :: PropsModifier ButtonProps
+secondary =
+  propsModifier
+    _
+      { kind = Secondary
+      }
+
+linkStyle :: PropsModifier ButtonProps
+linkStyle =
+  propsModifier
+    _
+      { type = "link"
+      -- , kind = LinkButton
+      }
+
+invisibleSpace :: String
+invisibleSpace = fromCharArray $ Array.catMaybes [ fromCharCode 0x2063 ]
```

Comment:
I believe we already have `nbsp` in `Lumi.Components.Text`

Reply:
This is U+2063 INVISIBLE SEPARATOR, which is presumably different from U+00A0 NON-BREAKING SPACE... although I haven't been able to find any information on what U+2063 does, so a comment to explain what this is doing might be helpful.

Reply:
I'd recommend writing this as `"\x2063"` by the way.